### ISSUE TITLE
write.neurons should be more careful about InputFileName

### ISSUE THREAD

**Issue by jefferis** [jefferis/nat#394]
See https://github.com/jefferis/rcatmaid/issues/120, but this shouldn't happen.

```
> orns=read.neurons.catmaid("ORN")
> write.neurons(orns, dir='orns.zip')
Error in basename(n$InputFileName) : a character vector argument expected
```

**Comment by jefferis** [comment]
@SridharJagannathan marginally more involved issue for nat!

**Comment by jefferis** [comment]
Logic error is here:

https://github.com/jefferis/nat/blob/f4af0b1c831ec721d7a05526deca389bfe11b285/R/neuron-io.R#L647-L655

**Comment by jefferis** [comment]
Maybe we should also specify a default format?
```
> orns=catmaid::read.neurons.catmaid("ORN")
  |===================================================================================================================================| 100%
> write.neurons(orns, dir='orns.zip')
 Error in write.neuron(n, dir = thisdir, file = files[nn], format = format,  : 
  No file specified and neuron does not have an InputFileName 
```

**Comment by jefferis** [comment]
Whereas this is ok:
```
> write.neurons(orns, dir='orns.zip', format='swc')
  adding: 1132350.swc (deflated 59%)
  adding: 1132750.swc (deflated 59%)
  adding: 1256592.swc (deflated 59%)
  adding: 1256615.swc (deflated 59%)
 ...
```

**Comment by jefferis** [comment]
closed by #396
